platforms/bukkit/common/src/main/java/com/dfsek/terra/bukkit/listeners/CommonListener.java

```diff
@@ -17,13 +17,37 @@
 
 package com.dfsek.terra.bukkit.listeners;
 
+import com.dfsek.terra.api.Platform;
+import com.dfsek.terra.bukkit.hooks.MultiverseGeneratorPluginHook;
+
+import org.bukkit.event.EventHandler;
 import org.bukkit.event.Listener;
+import org.bukkit.event.server.PluginEnableEvent;
+import org.mvplugins.multiverse.core.MultiverseCoreApi;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 
 /**
  * Listener for events on all implementations.
  */
 public class CommonListener implements Listener {
-    public CommonListener() {
+    private static final Logger logger = LoggerFactory.getLogger(CommonListener.class);
+    private final Platform platform;
+
+    public CommonListener(Platform platform) {
+        this.platform = platform;
+    }
+
+    @EventHandler
+    public void onPluginEnable(PluginEnableEvent e) {
+        if(e.getPlugin().getName().equals("Multiverse-Core")) {
+            try {
+                MultiverseCoreApi.get().getGeneratorProvider()
+                    .registerGeneratorPlugin(new MultiverseGeneratorPluginHook(platform));
+            } catch(Exception ex) {
```

Comment:
Overly broad exception, should just be catching `IllegalStateException` (& also `NoClassDefFoundError`)

nit: `ex` -> `e` to fit the rest of the codebase

Reply:
But I just think Terra shouldn't error out if any issue occur with Multiverse itself, bcu its really just a "suggest if possible"

Reply:
according to the docs it will only ever throw `IllegalStateException`, and then also `NoClassDefFoundError` if the class can't be loaded ofc

Reply:
Yea sure, I can change it if you prefer that